src/pages/AdminDashboard/UserPromotion/UserPromotion.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { Container, Paper, Grid, Switch, Typography } from '@material-ui/core';
+
+import useEndpoint from 'hooks/useEndpoint';
+import { promoteUser } from 'domains/AdminDashboard/api/api';
+import { UserPromotionType } from 'domains/AdminDashboard/types';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        width: '100%',
+        height: '100%',
+    },
+    paper: {
+        padding: theme.spacing(2),
+    },
+}));
+
+interface Props {
+    promotionOptions: string[];
+    userData: UserPromotionType;
+    onChange?: () => void;
+}
+
+const UserPromotion = ({ promotionOptions, userData, onChange: cb }: Props) => {
+    const classes = useStyles();
+    const [user, setUser] = useState<UserPromotionType>(userData);
+    const [post] = useEndpoint(() => promoteUser(user, user._id), {
+        onSuccess: cb,
+    });
+
+    const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, checked } = event.target;
+        const copyUser = user;
+        const newStatus = user.status.map((x) => {
+            if (x.status === name) {
+                return {
+                    ...x,
+                    active: checked,
+                };
+            }
+            return x;
+        });
+
```

Comment:
Lets use something other than x, even something such as `_status` or `innerStatus` would be fine here I think.